### PR TITLE
add CommunityVault contract

### PR DESCRIPTION
**Pull request by 0xb337r007**
## Description

Describe the changes made in your pull request here.

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Added natspec comments?
- [x] Ran `forge snapshot`?
- [x] Ran `pnpm lint`?
- [x] Ran `forge test`?

### CODE REVIEW

**Review by 0x-r4bbit — COMMENTED** [review]
Great start!

File: test/CommunityVault.t.sol [hunk 1]
```diff
@@ -38,10 +46,178 @@ contract CommunityVaultTest is Test {
...
+    }
+
+    function test_revertifcalledbynonadmin() public {
+        assertEq(erc20Token.balanceOf(address(vault)), 10e18);
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = 1;
+        amounts[1] = 1;
+
+        vm.prank(accounts[0]);
```
Comment (0x-r4bbit) [inline test/CommunityVault.t.sol:86]: The test contract itself is going to be `msg.sender` by default, so no `vm.prank()` needed here.

File: test/CommunityVault.t.sol [hunk 2]
```diff
@@ -38,10 +46,178 @@ contract CommunityVaultTest is Test {
...
+        erc20Token.transfer(address(vault), 10e18);
+    }
+}
+
+contract TransferERC20ByNonAdminTest is CommunityVaultBaseERC20Test {
+    function setUp() public virtual override {
+        CommunityVaultBaseERC20Test.setUp();
+    }
+
+    function test_revertifcalledbynonadmin() public {
```
Comment (0x-r4bbit) [inline test/CommunityVault.t.sol:79]: I believe this wants to be camelcase?

**Review by 0x-r4bbit — COMMENTED** [review]
This LGTM, just some minor comments and I believe this code doesn't manage to get past the linter.

File: test/CommunityVault.t.sol (hunk 1 above)
Comment (0xb337r007) [reply test/CommunityVault.t.sol:78]: I would prefer to have it explicitly written in case we change the way we deploy in the future, wdyt?

Comment (0x-r4bbit) [reply test/CommunityVault.t.sol:78]: That's perfectly fine. Was not an actionable comment.

**Comment by 0x-r4bbit** [comment]
@0xb337r007 this LGTM!

I'd just ask you to squash the commits as we don't need to have the steps where stuff didn't get past the linter etc.

**Comment by 0xb337r007** [comment]
@0x-r4bbit cool I squashed and merged it

**Comment by 0xb337r007** [comment]
I did it directly here in UI

**Pull request merged**